### PR TITLE
[Form+SecurityBundle] Trigger deprecation for csrf_provider+intention options

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

ping @stof

### CODE REVIEW

File: src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginFactory.php [hunk 1]
```diff
@@ -48,8 +48,27 @@ public function addConfiguration(NodeDefinition $node)
...
+                    $v['csrf_token_generator'] = $v['csrf_provider'];
+                    unset($v['csrf_provider']);
+
+                    return $v;
+                })
+                ->end()
             ->children()
-                ->scalarNode('csrf_provider')->cannotBeEmpty()->end()
+                ->scalarNode('csrf_token_generator')->cannotBeEmpty()->end()
             ->end()
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginFactory.php]: ``intention`` should be renamed to ``csrf_token_id`` too

Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginFactory.php]: I removed the beforeNormalizer for "intention": there is not "intention" in this configuration section

File: src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/MainConfigurationTest.php [hunk 2]
```diff
@@ -74,8 +74,8 @@ public function testCsrfAliases()
             'firewalls' => array(
                 'stub' => array(
                     'logout' => array(
-                        'csrf_provider' => 'a_token_generator',
```
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/MainConfigurationTest.php]: I think we should keep a configuration with `csrf_provider` in a legacy test to ensure not to introduce BC breaks in `2.8` patch releases in the future.

Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/MainConfigurationTest.php]: @xabbuh can you do it please?

Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/Tests/DependencyInjection/MainConfigurationTest.php]: see #16718

**Pull request merged**